src/components/Card.tsx

```diff
@@ -77,14 +77,18 @@ export const Card = (item: History): JSX.Element => {
     );
   };
 
-  switch (item.murojaahType) {
-    case MurojaahType.Juz:
-      return BaseCard(JuzCard);
-    case MurojaahType.Surah:
-      return BaseCard(SurahCard);
-    case MurojaahType.Ayah:
-      return BaseCard(AyahCard);
-    default:
-      return <></>;
-  }
+  const MemoizedCard = useMemo(() => {
+    switch (item.murojaahType) {
+      case MurojaahType.Juz:
+        return BaseCard(JuzCard());
+      case MurojaahType.Surah:
+        return BaseCard(SurahCard());
+      case MurojaahType.Ayah:
+        return BaseCard(AyahCard());
+      default:
+        return <></>;
+    }
+  }, []);
```

Comment:
this will never be updated if no dependency stated. please add `item` or `item.murojaahType`